### PR TITLE
refactor: publishing some functions and constants to make them accesible from dependents like the Kurtosis package catalog

### PR DESCRIPTION
**Pull request by leoporoli**

### CODE REVIEW

File: server/catalog/catalog_provider.go [hunk 1]
```diff
@@ -30,16 +30,26 @@ func GetPackagesCatalog() (PackageCatalog, error) {
...
 	}
+
+	newPackagesCatalog, err := GetPackageCatalogFromYamlFileContent(fileContent)
+
+	return newPackagesCatalog, nil
+}
+
+// GetPackageCatalogFromYamlFileContent receives the kurtosis-package-catalog.yml file content and returns a PackageCatalog object
+// it's public because it's also used by the kurtosis-package-catalog project
+func GetPackageCatalogFromYamlFileContent(fileContent []byte) (PackageCatalog, error) {
```
Comment (leoporoli) [inline server/catalog/catalog_provider.go:41]: I need to use it in the 'kurtosis-package-catalog' repository

File: server/consts/consts.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package consts
+
+const (
+	DefaultKurtosisYamlFilename = "kurtosis.yml"
```
Comment (leoporoli) [inline server/consts/consts.go:4]: I need to use them in the 'kurtosis-package-catalog' repository

File: server/github/client.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package github
+
+import (
+	"context"
+	github_lib "github.com/google/go-github/v54/github"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+)
+
+func CreateGithubClient(ctx context.Context) (*github_lib.Client, error) {
```
Comment (leoporoli) [inline server/github/client.go:10]: I need to use them in the 'kurtosis-package-catalog' repository

File: server/catalog/package_data_from_catalog.go [hunk 4]
```diff
@@ -20,6 +20,9 @@ const (
 )
 
 type packageDataFromCatalog struct {
+	// repositoryOwner is the Kurtosis package name
+	packageName types.PackageName
```
Comment (tedim52) [inline server/catalog/package_data_from_catalog.go:24]: `s/repositoryOwner/packageName`

**Review by tedim52 — APPROVED** [review]

**Pull request merged**